Update operator --set help text

blhagadorn opened the pull request:
This PR was in the old operator location, https://github.com/istio/operator/pull/745 before it got moved over to istio/istio.  It fixes the issue raised here: https://github.com/istio/istio/issues/19999


[ ] Configuration Infrastructure
[ x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ x] User Experience
[ ] Developer Infrastructure

blhagadorn committed:
Update root.go

istio-testing commented:
Hi @blhagadorn. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

howardjohn committed:
Add short --skip-confirmation flag (#20206)

Also fix trivial output error:
https://github.com/istio/istio/issues/20188

ostromart committed:
Fix chart copying and generation for operator build (#20145)

* Fix chart copying and generation for operator build

* Remove unstable dir prefix

* Update goldens, fix test stability

* Re-enable test stability check

* Make gen

* Make gen

* Update release script

* Make gen using container

* Update default profile

* Update values schema

* Change value to kubernetes

* Make gen

* Update goldens

* Make gen after merge

* Update goldens

* Make gen after merge

ramaraochavali committed:
simplify version check and add more tests (#20121)

* simplify version check and add more tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* slight change in condition

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

sushicw committed:
Add basic bench testing for analyzers (#20127)

* initial basic analyzer benchmark

* Artificial data bench v1

* clean up

* Do articifial data benchmarks per-analyzer

* Split test setup&run so we can reset the timer after setup

rshriram committed:
Fix listeners when proxy capture mode is none (#20159)

* Fix listeners when proxy capture mode is none

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* test fixes

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

selmanj committed:
Check resources for analyze suppression codes (#20174)

* Check resources for analyze suppression codes

* Add debug message when suppressing

elfinhe committed:
Add various of tag format support in version check of istioctl upgrade (#20187)

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#20210)

ostromart committed:
Add operator init/remove commands (#20180)

ostromart committed:
Update README for operator (#20181)

* Update README for operator

* Update links

ostromart committed:
Clean up makefile (#20222)

howardjohn committed:
Remove option for unsafe regex in Pilot (#20221)

* Remove option for unsafe regex in Pilot

For https://github.com/istio/istio/issues/20219
For https://github.com/istio/istio/issues/17053

Note this does not solve some places we were accidentally still using
unsafe regex, Yangmin is fixing those.

* Format

howardjohn committed:
Fix catchall route detection with safe regex (#20224)

howardjohn committed:
Fix validation cert rotation (#20229)

* Fix validation cert rotation

When kubernetes updates a secretMount, it is not always (ever? not
exactly sure why this ever works) a "modify" event. Generally its a
move, since its a symlink we are looking at. This changes the file
watching to be more liberal, which seems aligned with the rest of our
file watchers.

Fixes https://github.com/istio/istio/issues/20147

* Add back nil check

* Fix space

bianpengyuan committed:
Remove retry of individual time series (#19896)

* remove sd adapter retry

* only retry deadline exceed and unavailable.

* lint

elfinhe committed:
Allow multiple ICPS's as istioctl manifest/upgrade/profile input (#20190)

* Allow multiple ICPS's as istioctl manifest/upgrade/profile input

* Fix lint.

* Fix unit tests.

lambdai committed:
Add istio-validation container in when cni is enabled (#20130)

* add istio-init container in when cni is enabled

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* sync a little inject template, helm and istioctl

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* rename: cni preinit container as istio-validation

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* address comment

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* istio-validation previlege setting follows noncni

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* honor Romain

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* update injection template in istio-discovery

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* makegen: operator

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* yaml

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* golden

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* make gen

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

vikaschoudhary16 committed:
Skip unused services while generate clusters for the gateways (#20124)

* Skip unused services while generate clusters for the gateways

* Add istio-system clusters to all gateways

* Avoid double computation of listerners and routes

* Alternate approach using virtualservices

elfinhe committed:
Operator post-merge: update the data/versions.yaml (#20211)

* Update the data/versions.yaml, and also update the built-in link to istio/istio repo

* Fix make gen

* Fix make gen

* Use built-in versions.yaml by default.

yangminzhu committed:
support deny action in authorization policy (#20139)

* add framework support to deny authorization policies

* add tests

ramaraochavali committed:
eds test flakyness (#20237)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

howardjohn committed:
Enable linter on operator/ and manifests/ (#20144)

* Fix lint errors for operator

* Fix installer lint

* Format

* Fix new lint errors

* Regen

howardjohn committed:
Stop using deprecated CORS policy (#20220)

* Stop using deprecated CORS policy

For https://github.com/istio/istio/issues/20219

Stop using the deprecated Envoy field, and implement the new cors policy
in virtual service we apparently added

* Format

bianpengyuan committed:
Fix bootstrap override (#20240)

* remove bootstrap

* add new carts

* test fix

stewartbutler committed:
Updating CNI SHA so we build the new CNI image. (#20245)

howardjohn committed:
Fix pilot readiness probe (#20215)

Somehow this got reverted?

ostromart committed:
Schema tests (#20218)

* Add values.yaml schema test and add a bunch of missing fields

* Fix cases where tag is a JSON number

* Make gen

howardjohn committed:
References types by unique identifier (GroupVersionKind) instead of just Kind throughout Istio (#20214)

* Building works

* pkg/ tests passsing

* All tests passing

* Optimize allocs

* Format

* Remove find by group and kind

* Fix deny policy

JimmyCYJ committed:
Fix flaky test (#20251)

* fix flaky test

* format

jasonwzm committed:
Update operator CRD file to sync with generated file (#20243)

* Update operator CRD file to sync with generated file

* run make gen

* fix testsmanifest-generate/output/all_on.yaml

* update crd to update golden file

lambdai committed:
iptables: Add override for iptables validation timeout (#20249)

* Add override for iptables validation timeout
and reset default value to 5 seconds

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* pick suggestion

Signed-off-by: Yuchen Dai <silentdai@gmail.com>
Co-Authored-By: Romain Lenglet <romain.lenglet@berabera.info>

* fix build and make gen

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

Co-authored-by: Romain Lenglet <romain.lenglet@berabera.info>

howardjohn committed:
Clean up deprecated query param matchers (#20223)

For https://github.com/istio/istio/pull/20221

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#20254)

yangminzhu committed:
support negative match (not_ fields) in authorization policy (#20192)

Commit:
Remove -d usage on xarg for MacOS support (#20266)

sushicw committed:
Revert permissions pre-check, re-add fixed `--timeout` for `istioctl analyze` (#20250)

* Unrevert timeout

* Fix timeout, remove permissions pre-check

howardjohn committed:
Disable MCP by default (#20241)

* Disable MCP by default

* Set controller mode to strict

* Add debugging

* Fix event handler type

* Format

* Use global mcp

* Format

richardwxn committed:
add k8s service path to operator (#20233)

* add k8s service path

* fix test

* update gen

Commit:
Update sidecar build to use linux output path (#20273)

sushicw committed:
add integ test for `istioctl analyze` timeout (#20045)

* add integ test for istioctl analyze timeout

* This test can run without kube env, remove the unnecessary constraint

howardjohn committed:
Get security integration tests running on istioctl (#20239)

* TEST: check security operator tests

* Turn off wrong uds path flag for istioctl

* Fix postsubmit

* Fix mount for ingress sds

* Update goldens

* Revert hacks

howardjohn committed:
Fix golden files (#20285)

These slipped in breaking postsubmit somehow

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#20278)

howardjohn committed:
Turn off control plane security by default (#20281)

Fixes https://github.com/istio/istio/issues/20279

yxue committed:
disable listener filter timeout for outbound listener if port name is defined as tls or https (#20248)

* move trafficDirection to the listener options

* disable timeout for tls and https port

* Revert "move trafficDirection to the listener options"

This reverts commit 6a1aec91107572de0faaf1f2a09352476a5e78ee.

* fix

* update comment

* ..

lei-tang committed:
Support validating first party JWT for platforms without third party JWT (#20179)

- Support validating first party JWT for platforms without third party JWT.
- Add the JWT policy configuration to Operator.
- Config grpc calling credentials.

howardjohn committed:
Remove certmanager integration (#20269)

* Remove certmanager integration

This has been deprecated and removed from all docs for the past 2
releases. It is untested and does not work anymore (due to deprecation
of this version of certmanager).

* Regen golden

howardjohn committed:
Disable policy deployment by default (#20213)

* Disable policy deployment by default

* Update goldens

* Fix tests

* Disable policy checks fix

* Fix default e2e values

elfinhe committed:
Prune renamed and merged components. (#20288)

lei-tang committed:
Accept CA endpoint that is not on 15010 or 15012 (#20287)

ramaraochavali committed:
move enableEDSDebounce to init (#20294)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ayj committed:
Enable the config validation webhook in istiod (3 of 3) (#19985)

* add config validation to pilot

* update manifests

* fix clusterrole permissions

* update operator test data

* Update tests/integration/galley/webhook/webhook_test.go

Co-Authored-By: John Howard <howardjohn@google.com>

* update test framework to wait on pilot instead of galley

* more test fixes

* use pilot webhook port

* fix cert load

* undo test changes

* remove unused var

* skip key/cert reload test. istiod doesn't support rotation.

* skip istioctl webhook test until its updated to work with istiod

* update webhook tests to support istiod and non-istiod cases

* fix istiod check

* check override values

* gencheck

Co-authored-by: John Howard <howardjohn@google.com>

howardjohn committed:
Support graceful shutdown in Citadel (#20297)

Today, citadel will never shutdown on SIGTERM, only SIGKILL. With
default settings, this means 30s to shutdown (termination grace period =
30s by default). This also impacts tests.

The root cause of this is that we have a couple things blocking
shutdown. First, we do have some attempt at catching the SIGTERM, but it
only actually applies after ~45min of time.Sleeping(). Instead, we
should allow the stop channel to interupt that sleep.

The other problem is that when the monitoring server closes, it will
write an err on an unbuffered channel. In the normal case where things
exit graceful, there is a race between exiting because the stop channel
is closed, and actually reading from the monitoring error channel. To
get around this, we make the channel a buffered channel so we can write
asyncronously. This way if there is a real error, we will still see it,
but on shutdown we don't block.

howardjohn committed:
Stopgap for Galley dashboards with istiod (#20302)

* Stopgap for Galley dashboards with istiod

Right now on Istiod the galley dashboard is not useful because there is
no galley. In the 1.5 timeframe it may not be feasible to refactor our
dashboards to be feature oriented rather than component (if we even
desire this). In the short term, we can make all of the metrics shared
between istiod and galley not have a job= label. This will allow it to
pick up either source of the metric. We cannot do this for the other
metrics, as they are generic like CPU, etc.

* gen

ramaraochavali committed:
ads: ensure ads refuses connections when caches are not synced (#20296)

* ensure ads refuses connections when caches are not synced

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* move log lines to go func

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

seflerZ committed:
Pilot MCP bug fixes (#20304)

* istio bug fixes

* added some comments

* fixed to pass the integration test

* go fmt

howardjohn committed:
Fix mixer policy tests (#20303)

Regression since https://github.com/istio/istio/pull/20213

We turned on policy, but disabled checks. Didn't notice since it only
runs in postsubmit.

howardjohn committed:
Fix mixer policy tests (#20303)

Regression since https://github.com/istio/istio/pull/20213

We turned on policy, but disabled checks. Didn't notice since it only
runs in postsubmit.

esnible committed:
istioctl describe: fixes, test case, remove deprecated code (#20169)

* istioctl describe: fixes, test case, remove deprecated code

* Remove unneeded subtest structure

* Remove unnecessary change in describe

* Match describe output by regexp, not string

* Additional output to test the tests

* Keep constructor signature

* Fix compilation problem

* Look for Envoy config metadata in Istio 1.5.x format

* 'make format'

* Ran 'gen'

esnible committed:
istioctl describe: fixes, test case, remove deprecated code (#20169)

* istioctl describe: fixes, test case, remove deprecated code

* Remove unneeded subtest structure

* Remove unnecessary change in describe

* Match describe output by regexp, not string

* Additional output to test the tests

* Keep constructor signature

* Fix compilation problem

* Look for Envoy config metadata in Istio 1.5.x format

* 'make format'

* Ran 'gen'

esnible committed:
istioctl describe: fixes, test case, remove deprecated code (#20169)

* istioctl describe: fixes, test case, remove deprecated code

* Remove unneeded subtest structure

* Remove unnecessary change in describe

* Match describe output by regexp, not string

* Additional output to test the tests

* Keep constructor signature

* Fix compilation problem

* Look for Envoy config metadata in Istio 1.5.x format

* 'make format'

* Ran 'gen'

lei-tang committed:
Set Citadel as the default provider for Pilot certificate (#20196)

* Set Citadel as the default provider for Pilot certificate

- As some platforms may not have k8s signing APIs,
set Citadel as the default provider for Pilot certificate.
- Integration tests when pilotCertProvider=citadel

* Fix the webhook config controller with the citadel provider

* Fix the file permission problem

Fix the file permission problem:
error	failed to create discovery service: grpcDNS: open ./var/run/secrets/self-signed-root.pem: permission denied

* Write the root cert to a directory

* use correct caBundle for mutating webhook

Co-authored-by: Jason Young <jasonyoung@google.com>

blhagadorn commented:
/retest

istio-testing commented:
@blhagadorn: Cannot trigger testing until a trusted user reviews the PR and leaves an `/ok-to-test` message.

<details>

In response to [this](https://github.com/istio/istio/pull/20204#issuecomment-576073764):

>/retest
>


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

howardjohn committed:
Tune pilot logs for high endpoint scales (#20309)

Right now if you have a lot of endpoints Pilot logs are extremely
spammy. We log every IP every time a new IP is added, which for high #
of endpoints is often 1000s and very often. Then we log 1000s of lines
about EDS incremental update. I think those belong in debug to make the
logs a bit more sane.

Commit:
fix unused operator init wait option. (#20306)

* fix unused operator init wait option.

* fix test.

esnible commented:
/ok-to-test

esnible commented on operator/cmd/mesh/root.go:
```diff
@@ -24,9 +24,10 @@ import (
-	SetFlagHelpStr = `Set a value in IstioOperator CustomResource. e.g. --set policy.enabled=true.
-Overrides the corresponding path value in the selected profile or passed through IstioOperator CR
-customization file`
+	SetFlagHelpStr = `Can be used to override a path in the IstioOperator API spec, e.g. to choose a profile
```
Simplify grammar: `Override an IstioOperator value, e.g. ...`

blhagadorn replied:
That rolls of the tongue much better! Done.

blhagadorn replied:
I added the `Example:` to the cobraCommands for _manifest-apply.go_, however was unable to find any related commands that make use of `--set`.  Let me know if there is anything else I'm missing.

esnible reviewed (changes requested):
Also, please add `Example:` section to _istio/operator/cmd/mesh/manifest-apply.go_ and related commands that make use of `--set` showing a few common examples based on https://istio.io/docs/setup/install/istioctl/

```
istioctl manifest apply  # installs the default profile on the current Kubernetes cluster context
istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true # install the default profile with customizations
istioctl manifest apply --set installPackagePath=~/istio-releases/istio-1.4.3/install/kubernetes/operator/charts # install from external charts
istioctl manifest apply --set profile=demo # install a different profile.

irisdingbj committed:
change operator deploy files to consume API and repo merge changes (#20327)

ramaraochavali committed:
server: refresh push context after cache sync (#20323)

* refresh push context after cache sync

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* pass env

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* call push after cache sync

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

blhagadorn committed:
Update root.go

istio-testing commented:
@blhagadorn: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-istioio-k8s-tests_istio | 2561cc672c23c438ea1b7092017663304303e273 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/integ-istioio-k8s-tests_istio/3547) | `/test integ-istioio-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

blhagadorn committed:
Add Example cobra command for manifest apply

blhagadorn commented:
/retest

istio-testing commented:
@blhagadorn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-istioio-k8s-tests_istio | 2561cc672c23c438ea1b7092017663304303e273 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/integ-istioio-k8s-tests_istio/3547) | `/test integ-istioio-k8s-tests_istio`
gencheck_istio | 16bafeb4a3f137e22dbc1fab06f7ba4f460d0ad3 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/gencheck_istio/5128) | `/test gencheck_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@blhagadorn: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-istioio-k8s-tests_istio | 2561cc672c23c438ea1b7092017663304303e273 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/integ-istioio-k8s-tests_istio/3547) | `/test integ-istioio-k8s-tests_istio`
gencheck_istio | 16bafeb4a3f137e22dbc1fab06f7ba4f460d0ad3 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/gencheck_istio/5130) | `/test gencheck_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

blhagadorn committed:
Merge branch 'master' of github.com:blhagadorn/istio

blhagadorn commented:
Made necessary changes, not sure about that `gencheck_istio` job failing (where do I run `make gen`, in the base directory it throws a bunch of failures)?

blhagadorn commented:
/test gencheck_istio

istio-testing commented:
@blhagadorn: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 16bafeb4a3f137e22dbc1fab06f7ba4f460d0ad3 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20204/gencheck_istio/5137) | `/test gencheck_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible commented:
I had huge amounts of problems with **gencheck_istio** on a PR that should have gone through on Friday.  I suspect something is wrong with it.

I was able to get my PR to pass using `BUILD_WITH_CONTAINER=1 make gen` which does the work inside a Docker container.  (I am on OSX, I am using Docker Desktop).  However, even after all of that, my PR didn't merge.  Let's not worry about gencheck right now.  Let's see if we can get another positive review of the change while the gencheck stuff gets sorted out elsewhere.

howardjohn commented:
@esnible your PR not merging was a github bug, completely unrelated to gencheck. Just like in your PR, the go formatter has not run. This has been a part of Istio's CI forever and has not changed recently.

blhagadorn committed:
Ran make gen

blhagadorn commented:
Going to close this, sorry.
